packages/transport/test/modules/account/IdentityController.test.ts

```diff
@@ -0,0 +1,49 @@
+import { IDatabaseConnection } from "@js-soft/docdb-access-abstractions";
+import { sleep } from "@js-soft/ts-utils";
+import { AccountController, IdentityDeletionProcessStatus, Transport } from "../../../src";
+import { TestUtil } from "../../testHelpers/TestUtil";
+
+describe("IdentityController", function () {
+    let connection: IDatabaseConnection;
+    let transport: Transport;
+    let account: AccountController;
+
+    beforeAll(async function () {
+        connection = await TestUtil.createDatabaseConnection();
+        transport = TestUtil.createTransport(connection);
+
+        await transport.init();
+
+        const accounts = await TestUtil.provideAccounts(transport, 1);
+        account = accounts[0];
+        await account.init();
+    });
+
+    afterAll(async function () {
+        await account.close();
+        await connection.close();
+    });
+
+    afterEach(async () => {
+        const approvedIdentityDeletionProcess = await account.identityDeletionProcess.getIdentityDeletionProcessByStatus(IdentityDeletionProcessStatus.Approved);
+        if (approvedIdentityDeletionProcess) await account.identityDeletionProcess.cancelIdentityDeletionProcess(approvedIdentityDeletionProcess.id.toString());
+    });
+
+    test("check deletion of Identity that is not deleted", async function () {
+        const result = await account.identity.checkDeletionOfIdentity();
+        expect(result.value.isDeleted).toBe(false);
+        expect(result.value.deletionDate).toBeUndefined();
+    });
+
+    test("check deletion of Identity having IdentityDeletionProcess with expired grace period", async function () {
+        const identityDeletionProcess = await account.identityDeletionProcess.initiateIdentityDeletionProcess();
+
+        await sleep(10000);
```

Comment:
10s is hard :S

Reply:
I can decrease the grace period. To be honest, I wanted to check the pipelines first if anything breaks. 😅

Reply:
WTB 0?

Reply:
Ah no, then we cannot cancel the idp

Reply:
I heard that there is some discussion to override the gracePeriod when creating an idp, would be awesome for this